flags.go

```diff
@@ -136,7 +136,8 @@ func newApp() (app *cli.App) {
 			cli.StringFlag{
 				Name:  "endpoint",
 				Value: "https://storage.googleapis.com:443",
-				Usage: "The endpoint to connect to.",
```

Comment:
Please add unit test for the same. E.g. correct format, Incorrect format.

Reply:
We can't add unit-test for that. Ideally, both behaviours are correct. I have notified to @tritone to check, if it's by design or need to fix.

In the end, this will be taken care during the retry integration test with the help of https://github.com/googleapis/storage-testbench.